xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/AbstractRegistrationPage.java

```diff
@@ -78,12 +78,14 @@ public void fillRegisterForm(final String firstName, final String lastName, fina
         if (StringUtils.isNotEmpty(username)) {
             map.put("xwikiname", username);
         }
-        if (StringUtils.isNotEmpty(password)) {
-            map.put("register_password", password);
-        }
+        // We invert the order of password fields fill to test that the validation of this second password is properly
+        // updated when the first password field is changed.
```

Comment:
I'm not a huge fan of this: the PO is not supposed to be a place where we test stuff. That being said we have TestUtils methods for registration so we're not supposed to use that method outside of tests for registration. So I guess it's ok.